rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1661,6 +1661,167 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
+				empty
+				),
+			empty, empty, 
+			Create_RepriceTermsChangeInstruction(
+				tradeState -> trade -> tradableProduct -> product -> contractualProduct,
+				haircutPercentage,
+				effectiveRepriceDate
+				),
+			empty, empty, empty, empty
+			)
+		]
+		
+		
+func Create_RepriceTermsChangeInstruction: <"Creates the relevant terms change primitive instruction object for repricing a contractual product, which consists in the same terms as the original contractual product but with a new collateral price and haircut that may also result in modifying the cash or collateral quantity.">
+	inputs:
+		contractualProduct ContractualProduct (1..1) <"The original contractual product to be repriced.">
+		newHaircutPercentage number (1..1) <"The new haircut value based on cash adjustment.">
+		effectiveRepriceDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
+	output:
+		termsChangeInstruction TermsChangeInstruction (1..1) <"The relevant primitive instruction for the repricing.">
+
+	set termsChangeInstruction -> product -> contractualProduct:
+		contractualProduct
+	set termsChangeInstruction -> product -> contractualProduct -> economicTerms -> collateral -> collateralProvisions -> eligibleCollateral -> criteria -> treatment ->
+		valuationTreatment -> haircutPercentage:
+		newHaircutPercentage
+	set termsChangeInstruction -> product -> contractualProduct -> economicTerms -> effectiveDate:
+		effectiveRepriceDate
+		
+func Qualify_Reprice: <"This qualification function for security finance reprice verifies that the product is a security finance transaction with asset payout and interest rate payout, and the price after is different than the price before.">
+	[qualification BusinessEvent]
+	inputs:
+		businessEvent BusinessEvent (1..1)
+	output: is_event boolean (1..1)
+	alias beforeTradePrice: businessEvent -> instruction only-element -> before -> trade -> tradableProduct -> tradeLot->priceQuantity
+	alias afterTradePrice: FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> tradeLot->priceQuantity
+
+	alias beforeEconomicterms: businessEvent -> instruction only-element -> before -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+	alias afterEconomicTerms: FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+
+	set is_event:
+		//check that this is a security finance transaction with cash and collateral
+		businessEvent -> after -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> assetPayout exists and
+		businessEvent -> after -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> interestRatePayout exists and
+		beforeTradePrice exists and
+		afterTradePrice exists and
+		afterTradePrice <> beforeTradePrice and //price after does not equal price after
```

Comment:
think the comment is incorrect and superfluous really as it is clear what is being tested